pkg/vsphere/tasks/waiter.go

```diff
@@ -246,25 +246,17 @@ func IsMethodDisabledError(err error) bool {
 // IsNotFoundError checks if a soap fault or vim fault is ManagementObjectNotFound error
 func IsNotFoundError(err error) bool {
 	if soap.IsVimFault(err) {
-		if _, ok := soap.ToVimFault(err).(*types.ManagedObjectNotFound); ok {
-			return true
-		}
-		if _, ok := soap.ToVimFault(err).(*types.ManagedObjectNotFoundFault); ok {
-			return true
-		}
-
-		return false
+		_, ok1 := soap.ToVimFault(err).(*types.ManagedObjectNotFound)
+		_, ok2 := soap.ToVimFault(err).(*types.ManagedObjectNotFoundFault)
+		return ok1 || ok2
 	}
 
 	if soap.IsSoapFault(err) {
-		if _, ok := soap.ToSoapFault(err).VimFault().(types.ManagedObjectNotFound); ok {
-			return true
-		}
-		if _, ok := soap.ToSoapFault(err).VimFault().(types.ManagedObjectNotFoundFault); ok {
-			return true
-		}
+		_, ok1 := soap.ToSoapFault(err).VimFault().(types.ManagedObjectNotFound)
+		_, ok2 := soap.ToSoapFault(err).VimFault().(types.ManagedObjectNotFoundFault)
 
-		return soap.ToSoapFault(err).String == "vim.fault.ManagedObjectNotFound" ||
+		// sometimes we get the correct fault but wrong type
```

Comment:
What does wrong type mean? Can you put an example here in a comment. Since string comparison looks like a hack that may not be necessary.

Reply:
Actually I'm not sure what would the wrong type be. This is a historic comment carried over from old code. I have some guesses but not sure at all and I haven't seen any reproduce case.

Reply:
Can we log whatever type that happened to be unrecognized on debug level with the keyword in the log message such as DEVNOTICE.